[Flaky Test] Test `TestPGHoardWithPG::test_pause_on_disk_full` is flaky

Here you can find an occurrence of the flaky test:
https://github.com/aiven/pghoard/runs/4443587994?check_suite_focus=true

```
=========================== short test summary info ============================
FAILED test/test_pghoard.py::TestPGHoardWithPG::test_pause_on_disk_full - Ass...
====== 1 failed, 144 passed, 21 skipped, 10 warnings in 203.04s (0:03:23) ======
make: *** [Makefile:13: unittest] Error 1
Error: Process completed with exit code 2.
```

This is the relevant error message if I read it correctly:
```
test/test_pghoard.py::TestPGHoard::test_handle_site
  /opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/_pytest/threadexception.py:75: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-357
  
  Traceback (most recent call last):
    File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/threading.py", line 932, in _bootstrap_inner
      self.run()
    File "/home/runner/work/pghoard/pghoard/pghoard/walreceiver.py", line 155, in run
      self._init_cursor()
    File "/home/runner/work/pghoard/pghoard/pghoard/walreceiver.py", line 65, in _init_cursor
      self.conn = psycopg2.connect(self.dsn, connection_factory=PhysicalReplicationConnection)
    File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/psycopg2/__init__.py", line 122, in connect
      conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
    File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/psycopg2/extras.py", line 511, in __init__
      super().__init__(*args, **kwargs)
  psycopg2.OperationalError: connection to server at "127.0.0.4", port 5432 failed: Connection refused
  	Is the server running on that host and accepting TCP/IP connections?
```

--- a/pghoard/receivexlog.py
+++ b/pghoard/receivexlog.py
@@ -61,7 +61,7 @@ class PGReceiveXLog(Thread):
         self.pid = proc.pid
         self.log.info("Started: %r, running as PID: %r", command, self.pid)
         while self.running:
-            rlist, _, _ = select.select([proc.stdout, proc.stderr], [], [], 1.0)
+            rlist, _, _ = select.select([proc.stdout, proc.stderr], [], [], min(1.0, self.disk_space_check_interval))
             for fd in rlist:
                 content = fd.read()
                 if content:

